### ISSUE TITLE
[BUG]: [Regression] Older Intel Macs: Crash on launch after v1.7.5849 (Memory Map failure)

### ISSUE THREAD

**Issue by shinra-electric** [PCSX2/pcsx2#11728]
### Describe the Bug

PCSX2 crashes on launch on older Intel Macs with the following error message: 

`Failed to map data memory at an acceptable location`

### Reproduction Steps

I tried updating PCSX2 on an old Intel MacBook Air, and it crashed at launch. I bisected the builds, with v1.7.5847 working fine and all builds from v1.7.5849 were broken. (There was no v1.7.5848 release). 

Because the crash was instant at launch, there were no logs. However I noticed that if I enable logging as well as the log window (in the previous build), an alert would pop up and I could get a screenshot of the log window. 

I tested on a 2015 MacBook Air:
2.2Ghz Core i7
Intel HD 6000

Also tested on a 2015 MacBook Pro with the same results: 
2.7 Ghz Core i5
Intel Iris Graphics 6100

(I should also note that they are both running Sonoma via OCLP, though this shouldn't be an issue since v1.7.5847 works fine). 

<img width="372" alt="Screenshot 2024-08-19 at 17 03 47" src="https://github.com/user-attachments/assets/e9025ca2-8f9d-4637-8f4c-9a45a571f44b">
<img width="818" alt="Screenshot 2024-08-19 at 17 04 05" src="https://github.com/user-attachments/assets/56a9a00b-0922-4187-8b53-adb3f5a2114b">
![Screenshot 2024-08-19 at 17 12 14](https://github.com/user-attachments/assets/9dbe1eb4-ff73-4faf-9b44-a6bd22d741f0)

### PCSX2 Revision

v1.7.5849

### Operating System

macOS 14 (Sonoma)

**Comment by ynazar1** [comment]
Maybe the MAC build isn't universal or using ARM only features/flags? Working here on M1 Max. Sonoma 10.16.1 I don't have any games to test on mac, but it starting UI perfectly fine.

**Comment by shinra-electric** [comment]
I can confirm that #11734 fixes this on my 2015 MBP 👍🏼

**Comment by shinra-electric** [comment]
> Excellent - have you run this on a self-compiled version with the patch/change included?

No, I used the CI build from the GitHub action for this PR.

**Comment by atarixle** [comment]
It works for me. MacBook Pro 2014, OCLP, Sonoma 14.6.1
Weirdly I didn't even get the error-message shown in the first post here

**Comment by jimjamyaha** [comment]
This fixes the issue for me. MacBook Pro 2012, OCLP, Sonoma 14.6.1.

**Comment by atarixle** [comment]
BTW the same change in Duckstation lets it crash too. The build before "use mach routines for VM mem ... something" work.

**Comment by Alius17** [comment]
Tested  versions until latest nightly 2.3.66 and confirm not working ( Intel i5 Macbook pro 2015, 16gb Ram, MacOS Sequoia; Intel Xeon x5690 Mac Pro 2010, 32gb Ram, RX580 8Gb Vram, MacOS Sequoia), same error occurs and app crashes instantly. Version 1.7.5847 works well in both computers. If you need help testing any version, would gladly do so.

**Comment by Alius17** [comment]
Just install Version 1.7.5847 from the same page, and dont update till further notice. Works fine if you don't update. Good luck!

**Comment by yayakotto** [comment]
Thanks man, so I tried that and all the app icon does when click to open it is that it bounces up and down on my dock like it's about to open but never does. then when It does stop bouncing it just stays still without officially opening. You ever run into that issue?

**Comment by Alius17** [comment]
No, but you have to give permission to the app (full disk access) an also, you can check another usa bios file, or if you are in sequoia, allow permission from third party apps. Hope this helps. If not maybe you could post your logfile to see what is the problem...

**Comment by yayakotto** [comment]
Yeah I just added it to have full disk access but still nothing. Can you tell me how to retrieve my log file? Apologies I'm not too savvy with the process.

**Comment by Mrlinkwii** [comment]
@shinra-electric can i ask the status of this on lasttes nightly

**Comment by shinra-electric** [comment]
PR 13667 doesn't help this issue.

[emulog.txt](https://github.com/user-attachments/files/20296890/emulog.txt)

**Comment by RTHPJM** [comment]
I have this issue too. It started recently for me. I've been happily using PCSX2 for some time now, therefore I wondered what I had recently changed. It turns out that I decided to use OpenCore Legacy Patcher (OCLP) to re-enable my Broadcom-based wireless. (Apple removed Kernel Extension support for older Broadcom chipsets in Sonoma - I think, maybe earlier). To do this OCLP requires a specific set of System Integrity Protection (SIP) flags to be set. 

These flags disable elements of SIP. Knowing this I have tested each bit-wise flag using the NVRAM variable csr-active-config. It turns out that all flags are okay except one. When the flag CSR_ALLOW_UNAUTHENTICATED_ROOT is set (bit position 11, from zero-based bit counting), PCSX2 fails to start with the console output of:

> [    0.2081] mach_vm_map() failed: 3
> [    0.2081] mach_vm_map() failed: 3
> [    0.2081] mach_vm_map() failed: 3
> [    0.2081] mach_vm_map() failed: 3
> [    0.2081] mach_vm_map() failed: 1
> [    0.2081] mach_vm_map() failed: 1
> [    0.2081] mach_vm_map() failed: 1
> [    0.2081] mach_vm_map() failed: 1
> [    0.2081] mach_vm_map() failed: 1
> [    0.2081] mach_vm_map() failed: 1
> [    0.2081] mach_vm_map() failed: 1
> [    0.2083] ReportErrorAsync: Error: Failed to map data memory at an acceptable location.
> [    0.2083] ReportErrorAsync: Error: Failed to allocate VM memory.
> [    0.2092] Releasing host memory for virtual systems...


The CSR_ALLOW_UNAUTHENTICATED_ROOT flag is required by the OCLP team when re-enabling Broadcom wireless cards because the process performs root-patching. Root-patching means that the sealed System partition needs to be altered, hence the "root" System partition can no longer be validated as unmodified, thus requiring SIP to be downgraded.

At the moment then, I can have either WiFi enabled or PCSX2 usable, but not both. As to why PCSX2 is sensitive to this particular flag, I don't know! I will continue to research...

I'm posting here with the hope that @TellowKrinkle and the dev team might find this information useful.

References:
[Apple's csr.h header file from their Open Source Software repository](https://github.com/apple-oss-distributions/xnu/blob/main/bsd/sys/csr.h)
[OCLP Post-Installation manual page about SIP and the features re-enabled by root-patching](https://dortania.github.io/OpenCore-Legacy-Patcher/POST-INSTALL.html#sip-settings)
